core/src/main/java/nu/yona/server/messaging/service/MessageService.java

```diff
@@ -18,9 +20,9 @@
 import org.springframework.stereotype.Component;
 import org.springframework.stereotype.Service;
 
+import nu.yona.server.crypto.PublicKeyEncryptor;
```

Comment:
I don't think that is useful. Entities should not contain too much logic, logic should take place in the services.